src/AnalyticsEventService.ts

```diff
@@ -0,0 +1,25 @@
+import {EventAPIResponse} from "./EventAPIResponse";
+import {EventPayload} from "./EventPayload";
+
+export interface AnalyticsEventService {
+    /**
+     * Creates a new AnalyticsEventService with the following values defined
+     * as the default when making a report.
+     *
+     * @param payload - desired values to be applied. The new payload will
+     * override any overlapping values.
+     */
+    with(payload: EventPayload): AnalyticsEventService;
+
+    /**
+     * Reports an analytics event. Operand will throw an error if the request
+     * fails. The request will be sent via Beacon API so that requests are
```

Comment:
Given our recent discussion i think needs to be reworded

Reply:
Reworded based on thread: https://yext.slack.com/archives/G01SSAYJVM4/p1692731460458789?thread_ts=1692715631.720839&cid=G01SSAYJVM4